app/controllers/foreman_ansible/api/v2/hostgroups_controller_extensions.rb

```diff
@@ -45,10 +45,36 @@ def multiple_play_roles
 
             render_message @result
           end
+
+          api :POST, '/hostgroups/:id/play_ad_hoc_role',
+              N_('Plays an Ansible role ad hoc')
+          param :id, :identifier, :required => true
+          param :the_role_id, String, :required => true
+
+          def play_ad_hoc_role
+            # FIXME: When using just role_id, find resource will throw an
+            # exception: "undefined method `name' for nil:NilClass"
+            find_resource
+            find_ansible_role(params.require(:the_role_id))
```

Comment:
You can use [parameter_filter](https://github.com/theforeman/foreman_discovery/blob/develop/lib/foreman_discovery/engine.rb#L159-L162) to allow parameters you need